### ISSUE TITLE
f.e21.FWscHIST.ne30_L32_cam6_3_035.tphysac_reorder.001.hf2

### ISSUE THREAD

**Issue by cecilehannay** [NCAR/amwg_dev#16]
**Description:**
L32 + reordering

zmconv_parcel_pbl = .false.

**Case directory:**
Locally (if available):
/glade/p/cesmdata/cseg/runs/cesm2_0/f.e21.FWscHIST.ne30_L32_cam6_3_035.tphysac_reorder.001.hf2

On github (if available):
https://github.com/NCAR/amwg_dev/tree/f.e21.FWscHIST.ne30_L32_cam6_3_035.tphysac_reorder.001.hf2

**Sandbox:**
Locally (if still available):
/glade/work/hannay/cesm_tags/cam6_3_035.tphysac

On github (if available):
https://github.com/NCAR/amwg_dev/tree/cam6_3_035.tphysac/

**Diagnostics:**
AMWG diags (if available)
[https://webext.cgd.ucar.edu/FWscHIST/f.e21.FWscHIST.ne30_L32_cam6_3_035.tphysac_reorder.001.hf2/atm/](https://webext.cgd.ucar.edu/FWscHIST/f.e21.FWscHIST.ne30_L32_cam6_3_035.tphysac_reorder.001.hf2/atm/)

**Contacts:**
@adamrher 
@cecilehannay

**Comment by cecilehannay** [comment]
@adamrher 

Do you want me to keep the output  'KVH_CLUBB'. 
Will it work without the clubb diffusion coefficient in model_flags.F90?  Thanks.

**Comment by adamrher** [comment]
KVH_CLUBB will work anytime clubb is on, regardless of the model_flags changes. So I'd just keep it in the h0 files for now.

**Comment by cecilehannay** [comment]
@adamrher 
I am looking at difference between the new sandbox and the old sandbox in a L32. The difference are quite large. Is it expected? 
https://webext.cgd.ucar.edu/FWscHIST/f.e21.FWscHIST.ne30_L32_cam6_3_035.tphysac_reorder.001.hf2/atm/f.e21.FWscHIST.ne30_L32_cam6_3_035.tphysac_reorder.001.hf2.1980_1989-f.e21.FWscHIST.ne30_L32_cam6_3_019_plus_CESM2.2.001.hf.1980_1989/

**Comment by adamrher** [comment]
Yes, these SWCF differences look consistent with turning the trapezoidal rules off. [See here a run a did a while back w/ and w/o these off](https://webext.cgd.ucar.edu/F2000climo/cam6_3_006_F2000climo_f09_f09_mg17_1800pes_210324_postipdf_6macmic_notrap_Nx5yrs/atm/cam6_3_006_F2000climo_f09_f09_mg17_1800pes_210324_postipdf_6macmic_notrap_Nx5yrs.2_10-cam6_3_006_F2000climo_f09_f09_mg17_1800pes_201210_postipdf-6macmic-Nx10yrs.2_10/set5_6/set5_ANN_SWCF_c.png).

Might be worth doing a diff between the new trap run compared with zm2 run using the new code base, to see if the differences are consistent.

**Comment by swrneale** [comment]
Let's just change the one line in ZM back to the only 019 sandbox versions which I think increases the humidity.
I modified zm_conv.F90 from (https://github.com/NCAR/amwg_dev/tree/cam6_3_035.tphysac). I'll email it as git won't let me upload a F90 file !

**Comment by cecilehannay** [comment]
Hi @adamrher:
I ran the [diff between the new trap run compared with zm2 run using the new code base](https://webext.cgd.ucar.edu/FWscHIST/f.e21.FWscHIST.ne30_L48_BL10_cam6_3_035.tphysac_reorder_zm2_trapT.001.hf2/atm/f.e21.FWscHIST.ne30_L48_BL10_cam6_3_035.tphysac_reorder_zm2_trapT.001.hf2.1980_1989-f.e21.FWscHIST.ne30_L48_BL10_cam6_3_035.tphysac_reorder_zm2.001.hf2.1980_1989/). The differences look larger than in the [L32 new code/old code. ](https://webext.cgd.ucar.edu/FWscHIST/f.e21.FWscHIST.ne30_L32_cam6_3_035.tphysac_reorder.001.hf2/atm/f.e21.FWscHIST.ne30_L32_cam6_3_035.tphysac_reorder.001.hf2.1980_1989-f.e21.FWscHIST.ne30_L32_cam6_3_019_plus_CESM2.2.001.hf.1980_1989/)

**Comment by JulioTBacmeister** [comment]
So new sandbox = reorder+trapezoidal+ZM2, correct? Anything else?

**Comment by adamrher** [comment]
> I ran the diff between the new trap run compared with zm2 run using the new code base. The differences look larger than in the L32 new code/old code.

yeah I agree. it seems the new code *and/or* L58 is less sensitive to trapezoidal rules.

> So new sandbox = reorder+trapezoidal+ZM2, correct? Anything else?

Should be. Can we do a diff of atm_in for the L32 new and old code run, to be sure?

**Comment by swrneale** [comment]
There will be lots of difference because the original zm2 code has namelist controlled parcel energy mods, but they are turned off. Never really intended the old version to have that much longevity!

**Comment by JulioTBacmeister** [comment]
@cecilehannay The added diffusion mods will probably take a few more hours of work.  Won't have them today.

**Comment by adamrher** [comment]
> I ran the diff between the new trap run compared with zm2 run using the new code base. The differences look larger than in the L32 new code/old code.

I misread this earlier. It looks to me like the differences (L58+zm2 trap vs. no trap) are *smaller* than the differences in L32 new code/old code. Is that a typo or am I not looking at the right diagnostics?

**Comment by cecilehannay** [comment]
Sorry I meant: "The differences look larger in the L32 new code/old code" and not:
"The differences look larger **than** in the L32 new code/old code".

**Comment by adamrher** [comment]
> yeah I agree. it seems the new code and/or L58 is less sensitive to trapezoidal rules.

This actually makes sense. The trap rules should matter less for higher vertical resolution. They are by design supposed to increase accuracy for coarse vertical resolutions.
